tools/cpp/osx_cc_configure.bzl

```diff
@@ -157,17 +157,19 @@ def configure_osx_toolchain(repository_ctx, cpu_value, overriden_tools):
 
     env = repository_ctx.os.environ
     should_use_xcode = "BAZEL_USE_XCODE_TOOLCHAIN" in env and env["BAZEL_USE_XCODE_TOOLCHAIN"] == "1"
+    skip_xcode_fetch = "BAZEL_SKIP_XCODE_FETCH" in env and env["BAZEL_SKIP_XCODE_FETCH"] == "1"
     xcode_toolchains = []
 
     # Make the following logic in sync with //tools/cpp:cc_configure.bzl#cc_autoconf_toolchains_impl
-    (xcode_toolchains, xcodeloc_err) = run_xcode_locator(
-        repository_ctx,
-        paths["@bazel_tools//tools/osx:xcode_locator.m"],
-    )
-    if should_use_xcode and not xcode_toolchains:
-        fail("BAZEL_USE_XCODE_TOOLCHAIN is set to 1 but Bazel couldn't find Xcode installed on the " +
-             "system. Verify that 'xcode-select -p' is correct.")
-    if xcode_toolchains:
+    if not skip_xcode_fetch:
+        (xcode_toolchains, xcodeloc_err) = run_xcode_locator(
```

Comment:
There is a reference to xcodeloc_err on line 231 that caused an error when I tested it.  Is it reasonable to delete that and just fail hard here if there is an error?

Reply:
Trace the error handling back to https://github.com/bazelbuild/bazel/commit/cc21998c299b4d1f97df37b961552ff8168da17f#diff-c731a543f25a190052c3f045e894da8088b94ca4b0e067381fe281a46f26f3e0R671.  It's unfortunate the error handling grew farther and farther away over time.  I still don't understand if there are reasons why we don't just fail fast.

This may also be relevant: https://github.com/bazelbuild/bazel/pull/7519.